test/integration/discord.test.js

```diff
@@ -0,0 +1,104 @@
+const bot = require("../utils/generateBotToken");
+const { BAD_TOKEN, CLOUDFLARE_WORKER } = require("../../constants/bot");
+
+chai.use(chaiHttp);
+
+describe("Discord", function () {
+  let jwtToken;
+
+  beforeEach(async function () {
+    jwtToken = bot.generateToken({ name: CLOUDFLARE_WORKER });
+  });
+
+  afterEach(async function () {
+    await cleanDb();
+  });
+
+  describe("POST /discord", function () {
+    it("Should create a new discord data in firestore", function (done) {
+      chai
+        .request(app)
+        .post("/discord")
+        .set("Authorization", `Bearer ${jwtToken}`)
+        .send(discordData[0])
+        .end((err, res) => {
+          if (err) {
+            return done(err);
+          }
+          expect(res).to.have.status(201);
+          expect(res.body).to.have.property("message");
+          expect(res.body.message).to.equal("Added discord data successfully");
+
+          return done();
+        });
+    });
+
+    it("Should return 400 when adding incorrect discord data in firestore", function (done) {
+      chai
+        .request(app)
+        .post("/discord")
+        .set("Authorization", `Bearer ${jwtToken}`)
+        .send(discordData[1])
+        .end((err, res) => {
+          if (err) {
+            return done(err);
+          }
+
+          expect(res).to.have.status(400);
+          expect(res.body).to.have.property("message");
+          expect(res.body).to.have.property("error");
+          expect(res.body.message).to.equal('"linkStatus" must be a boolean');
+          expect(res.body.error).to.equal("Bad Request");
+
+          return done();
+        });
+    });
+
+    it("Should return 400 when authorization header is not present", function (done) {
+      chai
+        .request(app)
+        .post("/discord")
+        .send(discordData[0])
+        .end((err, res) => {
+          if (err) {
+            return done(err);
+          }
+
+          expect(res).to.have.status(400);
+          expect(res.body).to.have.property("message");
+          expect(res.body).to.have.property("error");
+          expect(res.body.message).to.equal("Invalid Request");
+          expect(res.body.error).to.equal("Bad Request");
+
+          return done();
+        });
+    });
+
+    it("Should return 401 when authorization header is incorrect", function (done) {
+      chai
+        .request(app)
+        .post("/discord")
+        .set("Authorization", `Bearer ${BAD_TOKEN}`)
+        .send(discordData[0])
+        .end((err, res) => {
+          if (err) {
+            return done(err);
+          }
+
+          expect(res).to.have.status(401);
+          expect(res.body).to.have.property("message");
+          expect(res.body).to.have.property("error");
+          expect(res.body.message).to.equal("Unauthorized Bot");
+          expect(res.body.error).to.equal("Unauthorized");
+
+          return done();
+        });
+    });
+  });
+});
```

Comment:
add a test for condition if the data is already present.

Reply:
This can't be tested as the data will be unique everytime because of the timestamps.